Sources/NIOHTTP2/HTTP2PipelineHelpers.swift

```diff
@@ -439,6 +439,7 @@ extension Channel {
     ///   - configuration: The settings that will be used when establishing the connection and new streams.
     ///   - position: The position in the pipeline into which to insert this handler.
     ///   - inboundStreamInitializer: A closure that will be called whenever the remote peer initiates a new stream.
+    ///     It must output the stream `Channel` (either directly or wrapped in some form) to allow users to iterate over inbound streams.
```

Comment:
This comment confuses me on all the methods here. The users can return whatever they want inside the initializer right. They might multiplex another `HTTP2` connection side a single stream and the return will be a `NIOHTTP2Handler.AsyncStreamMultiplexer<NIOHTTP2Handler.AsyncStreamMultiplexer<Output>>>`.

Additionally, we should make `Output: Sendable` here as well.

Reply:
You're right, this isn't correct, I was attempting to give guidance on the intent of the `Output` type but it's too narrow a description. I've reworded this to reflect the broader use.